python/ray/serve/api.py

```diff
@@ -1451,3 +1508,157 @@ def get_deployment_statuses() -> Dict[str, DeploymentStatusInfo]:
     """
 
     return internal_get_global_client().get_deployment_statuses()
+
+
+class ImmutableDeploymentDict(dict):
+    def __init__(self, deployments: List[Deployment]):
```

Comment:
Should this take in a dictionary or a list?

Reply:
this is an internal API so whichever you find easier in impl